### PR TITLE
refactor: use posix_spawn instead of NSTask so we can disclaim the spawned ShipIt executable

### PR DESCRIPTION
**Pull request by MarshallOfSound**
`posix_spawn` is nowhere near as pretty, but it does the job

Notes: no-notes

### CODE REVIEW

**Comment by deepak1556** [comment]
Possible to reuse [base::LaunchProcess](https://source.chromium.org/chromium/chromium/src/+/main:base/process/launch.h;l=234) instead, would require linking with `//base` but that might be a small price in favor of using a well maintained `posix_spawn` interface ?

File: patches/squirrel.mac/refactor_use_posix_spawn_instead_of_nstask_so_we_can_disclaim_the.patch [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+   ]
+   deps = [
+     ":mantle_framework+link",
+diff --git a/Squirrel/ShipIt-main.m b/Squirrel/ShipIt-main.mm
+similarity index 75%
+rename from Squirrel/ShipIt-main.m
+rename to Squirrel/ShipIt-main.mm
+index db246534e176f9c3ea2dd8b1c8659378fdc2435d..5fbcfb2fb9fa1346a7ed955dd2e32ce1c306b6d9 100644
+--- a/Squirrel/ShipIt-main.m
++++ b/Squirrel/ShipIt-main.mm
```
[the remark this answers is not in the document]
Comment (MarshallOfSound) [reply patches/squirrel.mac/refactor_use_posix_spawn_instead_of_nstask_so_we_can_disclaim_the.patch:28]: Uh, something complained, forgot which thing exactly

File: patches/squirrel.mac/refactor_use_posix_spawn_instead_of_nstask_so_we_can_disclaim_the.patch [hunk 2]
```diff
@@ -0,0 +1,135 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Samuel Attard <samuel.r.attard@gmail.com>
+Date: Mon, 28 Mar 2022 02:36:39 -0700
+Subject: refactor: use posix_spawn instead of NSTask so we can disclaim the
+ spawned ShipIt executable
+
+This ensures that if the ShipIt executable is hotswapped it doesn't inherit TCC permissions
+
```
[the remark this answers is not in the document]
Comment (MarshallOfSound) [reply patches/squirrel.mac/refactor_use_posix_spawn_instead_of_nstask_so_we_can_disclaim_the.patch:8]: As mentioned elsewhere, we are effectively the upstream, I will be sending this one back to the source repo though

File: patches/squirrel.mac/refactor_use_posix_spawn_instead_of_nstask_so_we_can_disclaim_the.patch [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+ 
+-							NSLog(@"Application launched at %@", bundleURL);
+-						}
++							pid_t pid = 0;
++
++							const char* launchPath = [exe fileSystemRepresentation];
++							const char* signal = [launchSignal fileSystemRepresentation];
++							const char* path = [bundleURL.path fileSystemRepresentation];
++							const char* args[] = { launchPath, signal, path, 0 };
++								int status = posix_spawn(&pid, [exe UTF8String], NULL, &attr, (char *const*)args, environ);
```
Comment (zcbenz) [inline patches/squirrel.mac/refactor_use_posix_spawn_instead_of_nstask_so_we_can_disclaim_the.patch:76]: Code and below have an extra padding.

File: patches/squirrel.mac/refactor_use_posix_spawn_instead_of_nstask_so_we_can_disclaim_the.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (nornagon) [reply patches/squirrel.mac/refactor_use_posix_spawn_instead_of_nstask_so_we_can_disclaim_the.patch]: yep, makes sense 👍🏻

**Review by zcbenz — APPROVED** [review]

**Comment by nornagon** [comment]
@deepak1556 I think using base from Squirrel is a bit odd as nothing in Squirrel uses base currently.

**Comment by deepak1556** [comment]
Yeah I agree 👍 . Also at the time of writing, my suggestion was based on building squirrel as part of electron but since we are also interested in pushing the changes back to https://github.com/Squirrel/Squirrel.Mac, building with base does not help.

**Pull request merged**